### PR TITLE
GDB-7867 Implement monitoring tab for query execution data

### PR DESCRIPTION
**Pull request by yordanalexandrov**
## What
Implement a new tab in the resource monitor view for query execution related charts and data

## Why
The view should help users to better evaluate the system when trying to solve problems

## How
- implemented necessary charts for the view
- added security for the new tab as only power user with admin right can see it

### CODE REVIEW

File: src/js/angular/resources/chart-models/resource/global-cache-chart.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            values: []
+        }];
+    }
+    addNewData(dataHolder, timestamp, data) {
+        const [hit, miss] = dataHolder;
+        hit.values.push([timestamp, data.cacheHit]);
+        miss.values.push([timestamp, data.cacheMiss]);
+    }
+    updateRange(dataHolder) {
+        const maxChartValue = Math.max(...dataHolder.filter((data)=> !data.disabled).flatMap((data) => data.values).flatMap((data) => data[1]));
```
Comment (svilenvelikov) [inline src/js/angular/resources/chart-models/resource/global-cache-chart.js:25]: This expression is repeated a couple of times. Can you pull it up somewhere in the base classes or in some utility?

File: src/js/angular/resources/controllers.js [hunk 2]
```diff
@@ -99,12 +128,114 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
...
+
+                    queryMonitorData = $timeout(getQueryMonitor, POLLING_INTERVAL);
+                });
+        };
+        const getPerformanceMonitorData = function () {
+            const activeRepository = $scope.getActiveRepository();
+            return MonitoringRestService.monitorQueryTransactionStatistics(activeRepository).then(function (response) {
+                return response.data;
+            }).catch(function (data) {
+                $scope.error = getError(data);
```
Comment (svilenvelikov) [inline src/js/angular/resources/controllers.js:176]: This is quite nasty way for error handling. Right now all request errors end up in this global error variable and they potentially can override each other. Can you centralize the logging by just handling it up the promise chain in a single place?

File: src/js/angular/resources/controllers.js [hunk 3]
```diff
@@ -99,12 +128,114 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
...
+
+        const getQueryMonitor = function() {
+            const activeRepository = $scope.getActiveRepository();
+            if ($scope.error) {
+                return;
+            } else if (!activeRepository) {
+                queryMonitorData = $timeout(getQueryMonitor, POLLING_INTERVAL);
+                return;
+            }
+            $q.all([getPerformanceMonitorData(), getActiveQueryMonitor()])
```
Comment (svilenvelikov) [inline src/js/angular/resources/controllers.js:141]: I'm not quite sure why you need to use the $q library for this? Doesn't Promise.all work here?

**Review by svilenvelikov — CHANGES_REQUESTED** [review]

File: src/js/angular/resources/chart-models/resource/global-cache-chart.js (hunk 1 above)
Comment (yordanalexandrov) [reply src/js/angular/resources/chart-models/resource/global-cache-chart.js]: Extracted it as a static method of the base class.

File: src/js/angular/resources/controllers.js (hunk 2 above)
Comment (yordanalexandrov) [reply src/js/angular/resources/controllers.js:172]: Will refactor error handling in a separate issue.

File: src/js/angular/resources/controllers.js (hunk 3 above)
Comment (yordanalexandrov) [reply src/js/angular/resources/controllers.js:137]: There is a problem with Promise API and angularJs

**Review by svilenvelikov — APPROVED** [review]

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Review by svilenvelikov — APPROVED** [review]

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Pull request merged**